dev/com.ibm.ws.config/test/com/ibm/ws/config/admin/internal/ConfigurationStorageHelperTest.java

```diff
@@ -0,0 +1,248 @@
+package com.ibm.ws.config.admin.internal;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.junit.Test;
+
+import com.ibm.ws.config.admin.ConfigID;
+import com.ibm.ws.config.admin.ConfigurationDictionary;
+import com.ibm.wsspi.kernel.service.utils.SerializableProtectedString;
+
+import junit.framework.AssertionFailedError;
+
+public class ConfigurationStorageHelperTest {
+    @Test
+    public void testStoreLoadCycle() throws IOException {
+        File configFile = new File("build", "testData");
+        ConfigurationDictionary testData = setupTestData();
+        String bundleLocation = "the location of the bundle";
+        Set<String> testVars = setupUniqueVars();
+        Set<ConfigID> testConfigID = setupConfigID();
+        ConfigurationStorageHelper.store(configFile, testData, bundleLocation, testConfigID, testVars);
+
+        Set<String> loadedUniqueVars = new HashSet<>();
+        Set<ConfigID> loadedConfigIDs = new HashSet<>();
+        ConfigurationDictionary loadedConfig = new ConfigurationDictionary();
+
+        String loadedLocation = ConfigurationStorageHelper.load(configFile, loadedUniqueVars, loadedConfigIDs, loadedConfig);
+
+        assertEquals("Location corrupted", bundleLocation, loadedLocation);
+        assertEquals("unique variables corrupted", testVars, loadedUniqueVars);
+        assertEquals("config ids corrupted", testConfigID, loadedConfigIDs);
+        assertConfigDictEquals(testData, loadedConfig);
+    }
+
+    @Test
+    public void testStoreLoadCycleWithNulls() throws IOException {
+        File configFile = new File("build", "testData");
+        ConfigurationDictionary testData = setupTestData();
+        ConfigurationStorageHelper.store(configFile, testData, null, null, null);
+
+        Set<String> loadedUniqueVars = new HashSet<>();
+        Set<ConfigID> loadedConfigIDs = new HashSet<>();
+        ConfigurationDictionary loadedConfig = new ConfigurationDictionary();
+
+        String loadedLocation = ConfigurationStorageHelper.load(configFile, loadedUniqueVars, loadedConfigIDs, loadedConfig);
+
+        assertNull("Location should be null", loadedLocation);
+        assertTrue("unique variables should be empty", loadedUniqueVars.isEmpty());
+        assertTrue("config ids should be empty", loadedConfigIDs.isEmpty());
+        assertConfigDictEquals(testData, loadedConfig);
+    }
+
+    @Test
+    public void testLoadExistingFile() throws IOException {
+        File configFile = new File("test-resources/test_config_cache", "testCache");
```

Comment:
Instead of checking binary testCache into git ... I would prefer this test store the expected results with ConfigurationStorageHelper.store and then load them back with ConfigurationStorageHelper.load and compare the results.